File: src/main/java/practice/adaptor/in/web/AccountController.java
package practice.adaptor.in.web;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import practice.account.application.in.CreateAccountUseCase;
import practice.account.application.in.TransactionUseCase;
import practice.common.dto.TransactionRequest;


@RestController
@RequiredArgsConstructor
public class AccountController {

  private final CreateAccountUseCase createAccountUseCase;
  private final TransactionUseCase transactionUseCase;

  @PostMapping("/account/saving")
  public void createAccount(@RequestParam("userId") Long userId) {
    createAccountUseCase.createSavingAccount(userId);
  }

  @PatchMapping("/account/saving/deposit")
  public void depositToSavingAccount(@RequestBody TransactionRequest transactionRequest) {
    transactionUseCase.depositToSavingAccount(transactionRequest.getUserId(), transactionRequest.getAmount());
  }
}

File: src/main/kotlin/practice/adaptor/in/web/AccountController.kt
package practice.adaptor.`in`.web

import org.springframework.web.bind.annotation.*
import practice.account.application.`in`.CreateAccountUseCase
import practice.account.application.`in`.TransactionUseCase
import practice.common.dto.CreateAccountRequest
import practice.common.dto.TransactionRequest

@RestController
@RequestMapping("/accounts")
class AccountController(
    private val createAccountUseCase: CreateAccountUseCase,
    private val transactionUseCase: TransactionUseCase
) {

    @PostMapping("/saving")
    fun createAccount(@RequestBody createAccountRequest: CreateAccountRequest) {
        createAccountUseCase.createSavingAccount(createAccountRequest.userId)
    }

    @PatchMapping("/saving/deposit")
    fun depositToSavingAccount(@RequestBody transactionRequest: TransactionRequest) {
        transactionUseCase.depositToSavingAccount(transactionRequest.userId, transactionRequest.amount)
    }
}